src/bit-systems/link-hover-menu.ts

```diff
@@ -1,47 +1,150 @@
-import { defineQuery, enterQuery, entityExists } from "bitecs";
+import { defineQuery, entityExists } from "bitecs";
 import type { HubsWorld } from "../app";
-import { Interacted, Link, LinkHoverMenu, LinkHoverMenuItem, HoveredRemoteRight } from "../bit-components";
-import { anyEntityWith } from "../utils/bit-utils";
+import { Link, LinkHoverMenu, HoveredRemoteRight, TextTag, Interacted, LinkHoverMenuItem } from "../bit-components";
+import { findAncestorWithComponent, findChildWithComponent } from "../utils/bit-utils";
+import { hubIdFromUrl } from "../utils/media-url-utils";
+import { Text as TroikaText } from "troika-three-text";
+import { handleExitTo2DInterstitial } from "../utils/vr-interstitial";
+import { changeHub } from "../change-hub";
+import { EntityID } from "../utils/networking-types";
+import { setMatrixWorld } from "../utils/three-utils";
+import { LinkType } from "../inflators/link";
 
-const NULL_EID = 0;
-
-const hoveredQuery = defineQuery([HoveredRemoteRight, Link]);
-const hoveredEnterQuery = enterQuery(hoveredQuery);
-const hoveredMenuItemQuery = defineQuery([HoveredRemoteRight, LinkHoverMenuItem]);
+const menuQuery = defineQuery([LinkHoverMenu]);
+const hoveredQuery = defineQuery([HoveredRemoteRight]);
 const clickedMenuItemQuery = defineQuery([Interacted, LinkHoverMenuItem]);
 
-export function linkHoverMenuSystem(world: HubsWorld) {
-  // Assumes always only single LinkHoverMenu entity exists for now.
-  // TODO: Take into account for more than one for VR
-  const menuEid = anyEntityWith(world, LinkHoverMenu)!;
-  const menuObject = world.eid2obj.get(menuEid)!;
-
-  // Save Link object eid in LinkHoverMenu when hovered.
-  hoveredEnterQuery(world).forEach((eid: number) => {
-    LinkHoverMenu.targetObjectRef[menuEid] = eid;
-    const targetObject = world.eid2obj.get(eid)!;
-    targetObject.add(menuObject);
-  });
+function updateLinkMenuTarget(world: HubsWorld, menu: EntityID, sceneIsFrozen: boolean) {
+  if (LinkHoverMenu.targetObjectRef[menu] && !entityExists(world, LinkHoverMenu.targetObjectRef[menu])) {
+    // Clear the invalid entity reference. (The link entity was removed).
+    LinkHoverMenu.targetObjectRef[menu] = 0;
+    return;
+  }
+
+  if (sceneIsFrozen) {
+    LinkHoverMenu.targetObjectRef[menu] = 0;
+    return;
+  }
+
+  const hovered = hoveredQuery(world);
+  const target = hovered.map(eid => findAncestorWithComponent(world, Link, eid))[0] || 0;
```

Comment:
I want to understand the reason why `Link` and `LinkHoverMenu` (below) is found from the ancestors because I'm not so familiar with our menu systems. Do we add `RemoteHoverTarget` component to descendant entities of `Link/LinkHoverMenu` entities?